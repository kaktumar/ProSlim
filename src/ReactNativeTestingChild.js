import React, {
  Component,
  PropTypes
} from 'react';
import {
  StyleSheet,
  View,
  Text,
  TouchableHighlight
} from 'react-native';

import { capitalizeWords } from './ReactNativeTestingUtils'

export default class ReactNativeTestingChild extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>
          Capitalized string:
        </Text>
        <Text style={styles.string}>
          {this._getCapitalizedString()}
        </Text>
        <TouchableHighlight onPress={this.props.onClear}>
          <Text style={styles.button}>
            Clear text
          </Text>
        </TouchableHighlight>
      </View>
    );
  }

  _getCapitalizedString () {
    const { text } = this.props;
    if (text) {
      return capitalizeWords(text);
    }

    return 'You must input something!';
  }
};

ReactNativeTestingChild.propTypes = {
  text: PropTypes.string.isRequired,
  onClear: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    alignItems: 'center'
  },
  title: {
    fontSize: 20,
    marginBottom: 20
  },
  string: {
    fontSize: 18,
    padding: 20
  },
  button: {
    marginTop: 30,
    fontSize: 30,
    color: '#007aff'
  }
});