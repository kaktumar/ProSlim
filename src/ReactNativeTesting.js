import React, { Component } from 'react';
import {
  StyleSheet,
  Dimensions,
  View,
  Text,
  TextInput,
  Image
} from 'react-native';
import Share from 'react-native-share';
import ReactNativeTestingChild from './ReactNativeTestingChild';

export default class ReactNativeTesting extends Component {
  constructor () {
    super();
    this.handleTextChange = this.handleTextChange.bind(this);
    this.clearText = this.clearText.bind(this);
    this.state = {
      text: ''
    };
  }

  render () {
    const { text } = this.state;
    return (
      <View style={styles.container}>
          <Image style={styles.logo} source={require('./react.png')} />
          <Text style={styles.title}>
              Welcome to React Native testing demo app
          </Text>
          <TextInput
              style={styles.input}
              placeholder={"write something"}
              onChangeText={this.handleTextChange}
              value={text}
          />
          <ReactNativeTestingChild text={text} onClear={this.clearText} />
      </View>
    );
  }

  handleTextChange (text) {
    this._setText(text);
  }

  clearText () {
    this._setText('');
  }

  _setText (text) {
    this.setState({text});
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    fontSize: 26,
    padding: 20,
    textAlign: 'center'
  },
  logo: {
    width: 50,
    height: 50
  },
  input: {
    width: (Dimensions.get('window').width - 20),
    height: 60,
    borderColor: 'gray',
    borderWidth: 1,
    borderRadius: 5,
    padding: 10,
    alignSelf: 'center'
  }
});
