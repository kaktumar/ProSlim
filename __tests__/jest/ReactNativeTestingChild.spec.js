jest.unmock('../../src/ReactNativeTestingChild');

import '../testutils/jest';
import ReactNativeTestingChild from '../../src/ReactNativeTestingChild';
import * as Utils from '../../src/ReactNativeTestingUtils';

describe('ReactNativeTestingChild', () => {
  let component;
  let onClearStub;
  const capitalizedString = 'mock string value';

  test('renders correctly', () => {
    const stringValue = 'random value';
    const snapshot = renderer.create(<ReactNativeTestingChild text={stringValue} onClear={onClearStub}  />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  beforeEach(() => {
    Utils.capitalizeWords = jest.fn(); // TODO: ???
    Utils.capitalizeWords.mockReturnValue(capitalizedString);
    onClearStub = jest.fn();
  });

  describe('when props.text is empty', () => {
    const stringValue = '';
    beforeEach(() => {
      component = shallow(<ReactNativeTestingChild text={stringValue} onClear={onClearStub} />);
    });

    it('renders placeholder string', () => {
      const expectedString = 'You must input something!';
      expect(component.find('Text').nodes[1].props.children).toEqual(expectedString);
    });

    it('does not call capitalizeWords', () => {
      expect(Utils.capitalizeWords.mock.calls.length).toBe(0);
    });

    describe('when clear text button is pressed', () => {
      it('should call onClear callback', () => {
        component.find('TouchableHighlight').simulate('press');
        expect(onClearStub).toBeCalled();
      });
    });
  });

  describe('when props.text is not empty', () => {
    const stringValue = 'random value';
    beforeEach(() => {
      component = shallow(<ReactNativeTestingChild text={stringValue} onClear={onClearStub} />);
    });

    it('renders string with capitalizeWords', () => {
      expect(Utils.capitalizeWords).toBeCalledWith(stringValue);
      expect(component.find('Text').nodes[1].props.children).toEqual(capitalizedString);
    });
  });
});