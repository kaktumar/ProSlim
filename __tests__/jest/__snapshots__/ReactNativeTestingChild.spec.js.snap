// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactNativeTestingChild renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 20,
      }
    }
  >
    Capitalized string:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "padding": 20,
      }
    }
  >
    mock string value
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#007aff",
          "fontSize": 30,
          "marginTop": 30,
        }
      }
    >
      Clear text
    </Text>
  </View>
</View>
`;
